### ISSUE TITLE
Stealth units and buildings are undetectable while in disabled state.

### ISSUE THREAD

**Issue by ThePredatorBG** [xezon/GeneralsGamePatch#238]
Stealth units and buildings are undetectable while in disabled state.

- Stuff detected before the EMP hits will remain visible.
- Stuff which was stealth at the time will remain hidden and unselectable (players can only see the EMP sparks on them).

There is a solution: EMP would reveal all disabled units and buildings on impact, so this bug fix is also going to affect gameplay, making the EMP more effective.
Example of effect on gameplay - a stealthed Scorpion hit by EMP will be undetectable and unattackable by units without damage radius such as Quads (force-firing being useless). If the proposed change is pushed, the Scorpion will lose its stealth when hit by the EMP and be targetable by Quads.
Discussion is needed.

I've attached some screenshots and a test map. There is an EMP Pulse power, which you have to launch at the enemy within the blue-textured borders. You have a Sentry Drone to test detection and a Quad Cannon to test against the Scorpion tank.

Test map: **[!EMP_SLTH_TEST.zip](https://github.com/xezon/GeneralsGamePatch/files/7122616/EMP_SLTH_TEST.zip)**

![shot_20210907_180927_3](https://user-images.githubusercontent.com/32432772/132370161-26b333c3-cce9-4e46-b5fb-3df2d8c9128b.jpg)
![shot_20210907_180857_2](https://user-images.githubusercontent.com/32432772/132370168-9c4a3bf4-ad77-45fd-b08b-c93951b3c81a.jpg)
![shot_20210907_175436_1](https://user-images.githubusercontent.com/32432772/132370172-fd149df1-51ce-4018-b14a-60a30fd70326.jpg)

**Comment by ThePredatorBG** [comment]
@xezon I think the issue should be labeled as a Bug too, since EMP makes stealth stuff immune to detection, which it makes no sense.

**Comment by Jundiyy** [comment]
Hmmm, this would also become a Spy Satellite for China, drop an EMP on a Scud Storm and also reveal everything around it? 
Although not as often so it won't be as revealing. Not sure on what just yet.

**Comment by commy2** [comment]
I don't think this argues for revealing buildings. The problem currently seems to be, that even an Outpost next to a building can not reveal it, if it's disabled.

**Comment by xezon** [comment]
This issue has been observed on stealthed USA Comanches, disabled by leaf let.

Dominator:

> 1. Stealth Commanches flying around (all upgrades)
> 2. leaflet used on his commanches, disables a tonne of them which landed on the floor
> 3. SOME of the commanches in black colour can now be seen with vees / drone and they get shot
> 4. However the REST glowing orange, they cannot be seen and cannot be shot, despite vee with drone being right there

## Replay

[00000000.zip](https://github.com/user-attachments/files/15867068/00000000.zip)

## Images

![post-199071-1718622424](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/4720891/25430f8a-3688-4762-9ce9-c436bb35ebaa)

![post-199071-1718622434](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/4720891/b49c9a4d-767d-4ba7-a278-280e5951c474)

![post-199071-1718622440](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/4720891/fdfac8e4-fd0e-48e3-bdb4-24043a1ac64b)

![post-199071-1718622445](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/4720891/a1b5058a-370f-4212-97fd-f2bfbe415e4e)
